[18.0][MIG] pos_order_to_sale_order: Migration to 18.0

benwillig opened the pull request.

legalsylvain committed:
[8.0] [ADD] pos order to sale order (w/o picking) (#188)

* [ADD] new module pos_order_to_sale_order

legalsylvain committed:
[FIX] taxes was dropped in the sale order lines. [IMP] Remove limitation of this module (price unit loss, and pricelist) if pos_pricelist is installed

legalsylvain committed:
[BUMP] version

Commit:
Coquille + ajout (BL) pour boutons "vente livrée"

pedrobaeza committed:
[FIX] por_order_to_sale_order: PEP8

oca-travis committed:
[UPD] Update pos_order_to_sale_order.pot

legalsylvain committed:
[FIX] do not allow to create sale order several times;
[FIX] Remove console.log ;
[IMP] Do not import js files in backend
[IMP] apply new readme structure
[FIX] remove useless demo file
[FIX] some pylint issues
[REF] add maintainers and developpment status

OCA-git-bot committed:
[UPD] README.rst

oca-travis committed:
[UPD] Update pos_order_to_sale_order.pot

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: pos-8.0/pos-8.0-pos_order_to_sale_order
Translate-URL: https://translation.odoo-community.org/projects/pos-8-0/pos-8-0-pos_order_to_sale_order/

legalsylvain committed:
[REF] pos_order_to_sale_order: Black python code

legalsylvain committed:
[MIG] pos_order_to_sale_order: Migration to 12.0

legalsylvain committed:
[FIX] remove commented code

Co-authored-by: Iván Todorovich <ivan.todorovich@druidoo.io>

oca-travis committed:
[UPD] Update pos_order_to_sale_order.pot

OCA-git-bot committed:
[UPD] README.rst

OCA-git-bot committed:
pos_order_to_sale_order 12.0.1.0.1

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: pos-12.0/pos-12.0-pos_order_to_sale_order
Translate-URL: https://translation.odoo-community.org/projects/pos-12-0/pos-12-0-pos_order_to_sale_order/

danimv5 committed:
Added translation using Weblate (Spanish)

danimv5 committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (23 of 23 strings)

Translation: pos-12.0/pos-12.0-pos_order_to_sale_order
Translate-URL: https://translation.odoo-community.org/projects/pos-12-0/pos-12-0-pos_order_to_sale_order/es/

augustodinizl committed:
Added translation using Weblate (Portuguese (Brazil))

augustodinizl committed:
Translated using Weblate (Portuguese (Brazil))

Currently translated at 100.0% (23 of 23 strings)

Translation: pos-12.0/pos-12.0-pos_order_to_sale_order
Translate-URL: https://translation.odoo-community.org/projects/pos-12-0/pos-12-0-pos_order_to_sale_order/pt_BR/

gaikaz committed:
pos_order_to_sale_order: Make JS extendable

Return module's javascript widgets to make them extendable for other modules.

Same idea as #535

OCA-git-bot committed:
pos_order_to_sale_order 12.0.1.0.2

legalsylvain committed:
[FIX] pos_order_to_sale_order : dependencies to sale module is now required in recent odoo version

legalsylvain committed:
[FIX] pos_order_to_sale_order : make the module working if get_order() is null, that occures if pos_restaurant is installed

oca-travis committed:
[UPD] Update pos_order_to_sale_order.pot

OCA-git-bot committed:
pos_order_to_sale_order 12.0.1.0.3

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: pos-12.0/pos-12.0-pos_order_to_sale_order
Translate-URL: https://translation.odoo-community.org/projects/pos-12-0/pos-12-0-pos_order_to_sale_order/

oca-travis committed:
[UPD] Update pos_order_to_sale_order.pot

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: pos-12.0/pos-12.0-pos_order_to_sale_order
Translate-URL: https://translation.odoo-community.org/projects/pos-12-0/pos-12-0-pos_order_to_sale_order/

manuelregidor committed:
[IMP] pos_order_to_sale_order: black, isort, prettier

manuelregidor committed:
[MIG]pos_order_to_sale_order: Migration to 13.0

oca-travis committed:
[UPD] Update pos_order_to_sale_order.pot

OCA-git-bot committed:
[UPD] README.rst

legalsylvain committed:
[PORT] pos_order_to_sale_order from 13.0 to 16.0:
- REF : Move pos.config configuration into res.config.settings
- REF : adapt JS and XML code to new OWL framework
- REF : Adapt python code to removal of onchange (simplification)
- IMP : Add action icons
- ADD : Add 'invoiced 'action
- IMP : Append PoS customer note in the sale order line name field
- IMP : Add tests

Commit:
[UPD] Update pos_order_to_sale_order.pot

OCA-git-bot committed:
[UPD] README.rst

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: pos-16.0/pos-16.0-pos_order_to_sale_order
Translate-URL: https://translation.odoo-community.org/projects/pos-16-0/pos-16-0-pos_order_to_sale_order/

Commit:
[UPD] Update pos_order_to_sale_order.pot

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: pos-16.0/pos-16.0-pos_order_to_sale_order
Translate-URL: https://translation.odoo-community.org/projects/pos-16-0/pos-16-0-pos_order_to_sale_order/

legalsylvain committed:
[FIX] pos_order_to_sale_order

set correct image path in the description of the module.

OCA-git-bot committed:
[UPD] README.rst

OCA-git-bot committed:
pos_order_to_sale_order 16.0.1.0.2

mymage committed:
Added translation using Weblate (Italian)

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (21 of 21 strings)

Translation: pos-16.0/pos-16.0-pos_order_to_sale_order
Translate-URL: https://translation.odoo-community.org/projects/pos-16-0/pos-16-0-pos_order_to_sale_order/it/

FLNerea committed:
Translated using Weblate (Spanish)

Currently translated at 66.6% (14 of 21 strings)

Translation: pos-16.0/pos-16.0-pos_order_to_sale_order
Translate-URL: https://translation.odoo-community.org/projects/pos-16-0/pos-16-0-pos_order_to_sale_order/es/

geomer198 committed:
[FIX] pos_order_to_sale_order: Added return for CreateOrderButton component.

OCA-git-bot committed:
pos_order_to_sale_order 16.0.1.0.3

GabbasovDinar committed:
[IMP] pos_order_to_sale_order: Improvement for the ability to use the result of sales order creation after inheritance

OCA-git-bot committed:
pos_order_to_sale_order 16.0.1.0.4

GabbasovDinar committed:
[16.0][REF] pos_order_to_sale_order: create method optimization

OCA-git-bot committed:
pos_order_to_sale_order 16.0.1.0.5

Ivorra78 committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (21 of 21 strings)

Translation: pos-16.0/pos-16.0-pos_order_to_sale_order
Translate-URL: https://translation.odoo-community.org/projects/pos-16-0/pos-16-0-pos_order_to_sale_order/es/

lbpatri committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (21 of 21 strings)

Translation: pos-16.0/pos-16.0-pos_order_to_sale_order
Translate-URL: https://translation.odoo-community.org/projects/pos-16-0/pos-16-0-pos_order_to_sale_order/es/

Ivorra78 committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (21 of 21 strings)

Translation: pos-16.0/pos-16.0-pos_order_to_sale_order
Translate-URL: https://translation.odoo-community.org/projects/pos-16-0/pos-16-0-pos_order_to_sale_order/es/

lbpatri committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (21 of 21 strings)

Translation: pos-16.0/pos-16.0-pos_order_to_sale_order
Translate-URL: https://translation.odoo-community.org/projects/pos-16-0/pos-16-0-pos_order_to_sale_order/es/

OCA-git-bot committed:
[UPD] README.rst

legalsylvain committed:
[FIX] pos_order_to_sale_order : make test compatible with pos_minimize_menu

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: pos-16.0/pos-16.0-pos_order_to_sale_order
Translate-URL: https://translation.odoo-community.org/projects/pos-16-0/pos-16-0-pos_order_to_sale_order/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (21 of 21 strings)

Translation: pos-16.0/pos-16.0-pos_order_to_sale_order
Translate-URL: https://translation.odoo-community.org/projects/pos-16-0/pos-16-0-pos_order_to_sale_order/it/

legalsylvain committed:
[REF] pos_order_to_sale_order : add sale order line sequence in the prepare function

legalsylvain committed:
[IMP] pos_order_to_sale_order : use customer_note to prepare sale order line name fields

legalsylvain committed:
[IMP] pos_order_to_sale_order : performance : directly overload _get_sale_order_line_multiline_description_sale to avoid to write on sale.order.line after having creating the sale.order.line

OCA-git-bot committed:
[BOT] post-merge updates

geomer198 committed:
[IMP] pos_order_to_sale_order: Code is refactored and tests are updated.

OCA-git-bot committed:
[BOT] post-merge updates

Commit:
Translated using Weblate (Portuguese (Brazil))

Currently translated at 100.0% (21 of 21 strings)

Translation: pos-16.0/pos-16.0-pos_order_to_sale_order
Translate-URL: https://translation.odoo-community.org/projects/pos-16-0/pos-16-0-pos_order_to_sale_order/pt_BR/

ivantodorovich committed:
[IMP] pos_order_to_sale_order: use test helpers

OCA-git-bot committed:
[BOT] post-merge updates

chaule97 committed:
[IMP] pos_order_to_sale_order: pre-commit auto fixes

chaule97 committed:
[MIG] pos_order_to_sale_order: Migration to 17.0

Commit:
[UPD] Update pos_order_to_sale_order.pot

OCA-git-bot committed:
[BOT] post-merge updates

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: pos-17.0/pos-17.0-pos_order_to_sale_order
Translate-URL: https://translation.odoo-community.org/projects/pos-17-0/pos-17-0-pos_order_to_sale_order/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (22 of 22 strings)

Translation: pos-17.0/pos-17.0-pos_order_to_sale_order
Translate-URL: https://translation.odoo-community.org/projects/pos-17-0/pos-17-0-pos_order_to_sale_order/it/

mtbochaca committed:
Added translation using Weblate (Catalan)

mtbochaca committed:
Translated using Weblate (Catalan)

Currently translated at 100.0% (22 of 22 strings)

Translation: pos-17.0/pos-17.0-pos_order_to_sale_order
Translate-URL: https://translation.odoo-community.org/projects/pos-17-0/pos-17-0-pos_order_to_sale_order/ca/

danielduqma committed:
[FIX] pos_order_to_sale_order: recompute taxes

Recompute `sale.order` taxes, as received ones are the ones from
product, without fiscal position mapping applied.

OCA-git-bot committed:
[BOT] post-merge updates

benwillig committed:
[MIG] pos_order_to_sale_order: Migration to 18.0

anmarmo1 reviewed (approved):
Functional review looks good!

<img width="1103" height="563" alt="image" src="https://github.com/user-attachments/assets/bef179b1-09db-4159-90df-2d5f3d939ad1" />

<img width="894" height="643" alt="image" src="https://github.com/user-attachments/assets/d0e231a6-07c7-4dd7-9fd0-0f0988fef66d" />

<img width="1304" height="512" alt="image" src="https://github.com/user-attachments/assets/a6dfc845-0766-4caa-927d-ff221b613eee" />

lbarry-apsl reviewed (approved):
Tested on the runboat. LGTM

lbarry-apsl commented:
Hi @legalsylvain ,

Could this module be merged?

lbarry-apsl commented:
Hi @pedrobaeza ,

Sorry, can this be merged?

pedrobaeza commented:
/ocabot migration pos_order_to_sale_order
/ocabot merge nobump

lbarry-apsl commented:
Hi @pedrobaeza , could you relaunch the merge?

pedrobaeza commented:
/ocabot merge nobump

lbarry-apsl commented:
@pedrobaeza , It didn't work, do you know why?

pedrobaeza commented:
/ocabot merge nobump

lbarry-apsl commented:
@benwillig , I think the JS tests aren't passing, could you check it?

benwillig commented:
> @benwillig , I think the JS tests aren't passing, could you check it?

@lbarry-apsl tests with OCB seems to failed because of other modules, and it's in the case in other PR in 18

benwillig committed:
[CHG] pos_order_to_sale_order: store the pos session on the created sale order + changed the way to display buttons

benwillig commented:
I updated the styles of the button to match the default one 

<img width="978" height="439" alt="image" src="https://github.com/user-attachments/assets/fc6e7395-4697-4f08-8248-b0e138f40d17" />

Also added pos_session_id on sale.order

ivilata commented:
Thanks @benwillig for the PR updates! The following OCB tests do fail, though:

- `test_frontend.TestDivideOrderSummary.test_divide_order_summary`
- `test_frontend.TestLotScanning.test_scan_lot_number`
- `test_frontend.TestLotScanning.test_scan_to_input_lot_number`
- `test_frontend.TestDisplayOrderNumber.test_display_order_number`
- `test_frontend.TestDisplayTotalQty.test_display_total_qty`

However, all such errors are similar, for example under log line 145 for `test_divide_order_summary`:

```
2025-09-26 14:12:38,558 251 INFO odoo odoo.addons.base.models.ir_model: Access Denied by ACLs for operation: read, uid: 10, model: ir.module.module 
2025-09-26 14:12:38,562 251 ERROR odoo odoo.http: Exception during request handling. 
Traceback (most recent call last):
  File "/opt/odoo/addons/point_of_sale/models/pos_session.py", line 181, in load_data
    response[model] = self.env[model]._load_pos_data(response)
[…]
odoo.exceptions.AccessError: You are not allowed to access 'Module' (ir.module.module) records.

This operation is allowed for the following groups:
	- Administration/Settings

Contact your administrator to request access if necessary.

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
[…]
  File "/opt/odoo/addons/point_of_sale/models/pos_session.py", line 185, in load_data
    'fields': self.env[model]._load_pos_data_fields(response['pos.config']['data'][0]['id']),
TypeError: IrModuleModule._load_pos_data_fields() takes 1 positional argument but 2 were given
```

Then the browser-side script gets stuck and eventually killed.

The code under `PosSession.load_pos_data` has changed a bit. Maybe rebasing/force-pushing can fix the issue? :shrug: 

**Update 2025-12-02:** The PR is already on top of the latest commit, please disregard my suggestion to rebase/force-push.

ivilata commented:
The errors that I listed above can be pinpointed respectively to these modules: `pos_divide_order_summary`, `pos_lot_barcode`, `pos_display_order_number`, `pos_display_total_quantity`. They do also trigger when run with a brand new DB and no trace of this PR's code. The tests of the `pos_order_to_sale_order` module added by this PR do succeed when run in isolation on a new DB, however.

@pedrobaeza, should the other modules be fixed first, before this one can be merged? I understand that merging this one is kind of ugly as the "default" ensemble of tests run by `oca_run_tests` does not pass. Thanks!

pedrobaeza commented:
Which tests are failing here in GitHub?

fabiandumaguala reviewed (approved):
LGTM

pedrobaeza commented:
/ocabot merge nobump

ivilata commented:
@pedrobaeza: The tests that fail are the ones that I listed in https://github.com/OCA/pos/pull/1429#issuecomment-3528186982. They fail both in GitHub's CI and when run locally.

pedrobaeza commented:
But it's due to this module or they are failing in general?

ivilata commented:
The tests that I mentioned from modules `pos_divide_order_summary`, `pos_lot_barcode`, `pos_display_order_number`, `pos_display_total_quantity` are failing when I run them in a local Odoo 18 installation, even if I do not enable the `pos_order_to_sale_order` module from this PR, so I guess that the former ones do have issues of their own. When I run the tests only for `pos_order_to_sale_order` locally, they pass just right.

Sorry if I didn't explain myself more clearly. :slightly_smiling_face:

pedrobaeza commented:
Then they should be fixed before merging this.

ivilata commented:
@pedrobaeza @benwillig: I traced back the errors in the other modules to a bug introduced in upstream Odoo 18 in commit https://github.com/odoo/odoo/commit/282a667bae54: a new `IrModuleModule._load_pos_data_fields(self)` method was introduced (then called) in module `point_of_sale` having only the `self` parameter, while code elsewhere in Odoo defines and uses `_load_pos_data_fields(self, config_id)` instead. Then `PosSession.load_data()` ended up calling it with the extra argument, which causes the errors.

The following changes fix the issue:

```diff
diff --git a/addons/point_of_sale/models/ir_module_module.py b/addons/point_of_sale/models/ir_module_module.py
index fcae0cad27a1..32c81bd36724 100644
--- a/addons/point_of_sale/models/ir_module_module.py
+++ b/addons/point_of_sale/models/ir_module_module.py
@@ -5,7 +5,7 @@ class IrModuleModule(models.Model):
     _inherit = 'ir.module.module'
 
     @api.model
-    def _load_pos_data_fields(self):
+    def _load_pos_data_fields(self, config_id):
         return ['id', 'name', 'state']
 
     @api.model
@@ -14,8 +14,8 @@ class IrModuleModule(models.Model):
 
     def _load_pos_data(self, data):
         domain = self._load_pos_data_domain()
-        fields = self._load_pos_data_fields()
+        fields = self._load_pos_data_fields(data['pos.config']['data'][0]['id'])
         return {
             'data': self.search_read(domain, fields, load=False),
-            'fields': self._load_pos_data_fields(),
+            'fields': fields,
         }
```

I don't have time right now to work on a new PR for that, if anyone wants to use the patch to create the PR, be my guest. Thanks! :slightly_smiling_face: 

(I may be wrong, but I the existing bug may be blocking all future PRs to this repo as well since it will make their tests fail.)

**Update (2026-02-17):** I did create the PR to upstream Odoo, see comment below.

niboo-ave commented:
Hello,

On my local I have a warning : Component 'CreateOrderButton' does not have a static props description

niboo-ave commented:
After some test we also notice that when a  pos order customer have a price list with discount, and that we create a SO from the pos: each SO have the unit price = to price with the discount and that discount is at False.
This seems to come from the POS order serialisation that is not precise enough.

I do not have any solution so far for this. But I guess it's worth it to let you know

niboo-ave commented:
> After some test we also notice that when a pos order customer have a price list with discount, and that we create a SO from the pos: each SO have the unit price = to price with the discount and that discount is at False. This seems to come from the POS order serialisation that is not precise enough.
> 
> I do not have any solution so far for this. But I guess it's worth it to let you know

After longer investigation, this make sens that odoo standard do it this way because it's easier to apply a discount per line that is not "cumulated".

ivilata commented:
@benwillig @pedrobaeza In case you're interested, I did create the PR to fix the issues that I mentioned above in `IrModuleModule._load_pos_data_fields()` in upstream Odoo. Let's hope they accept it, lest updates to the OCA/pos repos be no longer possible! :crossed_fingers:

pedrobaeza commented:
We already did it: https://github.com/odoo/odoo/pull/241122

ivilata commented:
Thanks @pedrobaeza! Since both PRs ended up being the same patch, I closed mine in favor of yours, which already has some positive reviews. Let's hope that @adgu-odoo (or @stva-odoo for that matter) can provide the required positive review so that we can unlock this! :slightly_smiling_face:

HaraldPanten commented:
@pedrobaeza Could you try merging this one again? This PR has been merged --> https://github.com/odoo/odoo/pull/241122

THX!

pedrobaeza commented:
/ocabot merge nobump
